internal/app/handle.go

```diff
@@ -36,19 +51,32 @@ func handleGokuRequests(w http.ResponseWriter, r *http.Request) {
 func handleGokuPost(w http.ResponseWriter, r *http.Request) {
 	// Reads the body of the JSON request into the fighter struct
 	// by unmarshaling the JSON bytes
-	f := fighter{}
+	f := fighters{}
 	body, err := ioutil.ReadAll(r.Body)
 	if err != nil {
 		log.Fatal(err)
 	}
 	json.Unmarshal(body, &f)
 
-	// Concatinates the name from the POST body with the string
+	// Concatinates the name from the POST body with a string
 	// then marshals the response back into JSON bytes to
 	// send to the browser.
-	a := allegiance{
-		Message: f.Name + " has joined the Z squad",
+	a := allegiance{}
+	switch length := len(f.Fighters); {
+	case length == 1:
+		lenOfOne(f.Fighters[0], &a)
```

Comment:
`lenOfOne` is the condition for triggering this function in this case, but what is the actual funcationaliy?